### ISSUE TITLE
if the connection from proxy to broker is down, it will cause many client connections to be closed 

### ISSUE THREAD

**Issue by hujie6** [streamnative/mop#675]
**Is your enhancement request related to a problem? Please describe.**
If one thousand devices are connected to the proxy node, and the connection pool size from proxy to broker is one. that means one thousand devices share one connection. if the connection from proxy to broker is closed due to some exceptions, it will cause 1000 devices to go offline at the same time. This may not be what the user expects to see.

**Describe the solution you'd like**
1. If an exception occurs from the proxy to the broker, we may not close the shared connection, or do not close the device client connection.

**Comment by Technoboy-** [comment]
In this case, you'd better set the pool size more than 1.

**Comment by hujie6** [comment]
> In this case, you'd better set the pool size more than 1.

Yes, in a production environment, the pool size must be greater than 1. but one of the connections is closed，it will also cause a large number of devices disconenction or connection.

**Comment by mattisonchao** [comment]
@hujie6 
That is a great idea, but we should consider how to process continuous publishing messages. (In some high-rate conditions).

**Comment by Technoboy-** [comment]
> > In this case, you'd better set the pool size more than 1.
> 
> Yes, in a production environment, the pool size must be greater than 1. but one of the connections is closed，it will also cause a large number of devices disconenction or connection.

yes, right, but reconnection will not have much impact on the business in the MQTT scenario.
